src/java/org/apache/cassandra/db/Columns.java

```diff
@@ -459,42 +466,110 @@ public String toString()
 
     public static class Serializer
     {
+        AbstractTypeSerializer typeSerializer = new AbstractTypeSerializer();
+
         public void serialize(Columns columns, DataOutputPlus out) throws IOException
         {
-            out.writeUnsignedVInt(columns.size());
+            int regularCount = 0;
+            int syntheticCount = 0;
+
+            // Count regular and synthetic columns
+            for (ColumnMetadata column : columns)
+            {
+                if (column.isSynthetic())
+                    syntheticCount++;
+                else
+                    regularCount++;
+            }
+
+            // Jam the two counts into a single value to avoid massive backwards compatibility issues
```

Comment:
The only compatibility issue is that we'll need to upgrade writers before coordinators, correct?

Reply:
Correct.